src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go

```diff
@@ -381,6 +369,103 @@ func (a *StorageMinerActorCode_I) ProcessVerifiedElectionPoSt(rt Runtime) InvocO
 	return a._onSuccessfulPoSt(rt)
 }
 
+// called by verifier to update miner state on successful surprise post
+// after it has been verified in the storage_mining_subsystem
+func (a *StorageMinerActorCode_I) ProcessSurprisePoSt(rt Runtime, onChainInfo sector.OnChainPoStVerifyInfo) InvocOutput {
+	TODO() // TODO: validate caller
+
+	if !a._verifySurprisePoSt(rt, onChainInfo) {
+		rt.Abort(
+			exitcode.UserDefinedError(exitcode.PoStVerificationFailed),
+			"PoSt verification failed")
+	}
+
+	// Ensure pledge collateral satisfied
+	// otherwise, abort ProcessVerifiedSurprisePoSt
+	a._ensurePledgeCollateralSatisfied(rt)
+
+	return a._onSuccessfulPoSt(rt)
+
+}
+
+func (a *StorageMinerActorCode_I) GetWorkerKeyByMinerAddress(minerAddr addr.Address) filcrypto.VRFPublicKey {
+	panic("TODO")
+}
```

Comment:
This doesn't make sense here, and it looks like you removed calls to it.

What would make sense is
```
func (a *SMAC_I) GetWorkerKey() {
    h, st := a.State(rt)
    w := st.Info().Worker
    Release(...)
    return w
}
```